### DISCUSSION TITLE
Introducing TaskwarriorGPT: an AI Assistant for Taskwarrior Users

### DISCUSSION THREAD

**Discussion by younes-io** [GothenburgBitFactory/taskwarrior#3200] [Ideas]
I discovered today this awesome CLI called `taskwarrior` after trying other similar tools, and I needed a way to learn very quickly, so I thought maybe I could leverage [GPTs](https://openai.com/blog/introducing-gpts) to create an assistant that would provide help for me.

It helped me discover how to write `task` commands, configure `~/.taskrc` for custom reports, themes, etc.

Link of the GPT: https://chat.openai.com/g/g-vHnl2dh7J-taskwarriorgpt

I posted this here to share with the community and to get your feedback.

BTW, @GothenburgBitFactory :
1 - Can I use the logo in the GPT ?
2 - Any questions or concerns / comments ?

EDIT:
3 - Could we add this GPT link to the README ?

---

Simple questions:
============
![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/3153107/84caf046-81cf-4a3c-b674-76a0c2fa4d29)

![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/3153107/7e3d59d8-8ed1-4d3e-8d3a-67358b13a553)

A bit less simple:
===========
![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/3153107/d17ef14d-67b9-45c3-a035-47e563c1ba6f)

![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/3153107/2720781e-d5f8-41cb-a1ae-a9633deca2cd)

**Comment by djmitche** [comment]
You're welcome to use the logo - we do not have a trademark registration or anything like that, but I appreciate the request!

This appears to require a paid account (at least, I assume that's what "ChatGPT Plus" is), and I'd rather not link to paid offerings in the README. I would suggest adding it to https://taskwarrior.org/tools/ but that can only link to GitHub repositories AIUI. @lauft do you have other ideas?

**Reply by younes-io** [reply]
Yes, the "GPTs" feature is available for paid accounts.
For the README, makes sense! 
Anyway, I'll keep an eye on the comments / feedback of users to improve and update this over time :)

**Reply by lauft** [reply]
As Taskwarrior/GothenburgBitFactory is not directly involved with this, I would also recommend against linking this in the project README.

> I would suggest adding it to https://taskwarrior.org/tools/ but that can only link to GitHub repositories AIUI

Linking GitHub repositories is what the automatic crawler script does (and only can do), but it is possible to manually add a tool to the page by adding something like the following to the [bin/include.json](https://github.com/GothenburgBitFactory/tw.org/blob/master/bin/include.json)
```
    {
      "description": "TaskwarriorGPT: an AI Assistant for Taskwarrior Users",
      "dormant": false,
      "language": [ "ChatGPT" ],
      "license": "unknown",
      "name": "TaskwarriorGPT",
      "owner": [
        "younes-io"
      ],
      "rating": null,
      "updated": "2023-11-16",
      "url": "https://chat.openai.com/g/g-vHnl2dh7J-taskwarriorgpt"
    }
```
@younes-io , please check whether above information is correct.

**Reply by younes-io** [reply]
I think that's OK

**Comment by eqjjh** [comment]
Did you upload any data to train this on or did it train itself, and in that case, which version, presumably 2.6.2?

**Reply by lauft** [reply]
👉🏻 @younes-io

**Reply by younes-io** [reply]
I uploaded the man docs, version 2.6.2

**Comment by djmitche** [comment]
https://github.com/pgibler/cmdh looks like a more general version of this

**Reply by younes-io** [reply]
@djmitche there are many tools like that (such as `sgpt`, etc), but those only use the model's general knowledge that the user configures in their terminal / env.
This taskworrior GPT is different: when a user asks a question, it performs information/document retrieval from the "man/doc" of taskwarrior, and is therefore more accurate.

**Reply by younes-io** [reply]
also, this GPT as I mentioned earlier, was developed as a way for me to learn taskwarrior quickly and get the most recent options/commands

**Comment by djmitche** [comment]
https://github.com/GothenburgBitFactory/taskwarrior/pull/3689 seems related to this work, FYI.
